pkg/controller/submariner/submariner_controller.go

```diff
@@ -214,6 +224,15 @@ func (r *ReconcileSubmariner) reconcileDaemonSet(owner metav1.Object, daemonSet
 	return errorutil.WithMessagef(err, "error creating or updating DaemonSet %s/%s", daemonSet.Namespace, daemonSet.Name)
 }
 
+func (r *ReconcileSubmariner) reconcileServiceDiscoverCR(submariner *submarinerv1alpha1.Submariner, reqLogger logr.Logger) error {
+	sd := newServiceDiscoveryCR(submariner)
+	_, err := r.submClientSet.SubmarinerV1alpha1().ServiceDiscoveries(submariner.Namespace).Create(sd)
+	if err != nil {
+		reqLogger.Error(err, "Error creating service discovery CR")
+	}
```

Comment:
It seems this should use CreateOrUpdate logic.

Reply:
Added update and retry.